app/src/main/java/com/example/pinterbest/LoginFragment.kt

```diff
@@ -24,19 +41,94 @@ class LoginFragment : Fragment() {
         super.onViewCreated(view, savedInstanceState)
 
+        initObservers(view)
+
         binding.loginButton.setOnClickListener {
-            if (isValidLogIn()) {
-                Toast.makeText(context, "Correct", Toast.LENGTH_SHORT).show()
-                it.findNavController().navigate(R.id.action_loginFragment_to_homeFragment)
-            } else {
-                Toast.makeText(context, "Not correct", Toast.LENGTH_SHORT).show()
-                it.findNavController().navigate(R.id.action_loginFragment_to_homeFragment)
+            if (validateUserFields()) {
+                model.setLiveEvent(getUserData())
+            }
+        }
+
+        binding.emailAuth.setOnClickListener {
+            it.findNavController().navigate(R.id.action_loginFragment_to_registrationFragment)
+        }
+    }
+
+    private fun initViewModels() {
+        model = ViewModelProvider(
+            requireActivity(),
+            LoginFactory(
+                requireActivity().application,
+                Repository(
+                    preferences = requireActivity().getSharedPreferences(
+                        getString(R.string.login_info),
+                        Context.MODE_PRIVATE
+                    )
+                )
+            )
+        ).get(LoginViewModel::class.java)
+    }
+
+    private fun initObservers(view: View) {
+        model.getLiveEvent().observe(
+            viewLifecycleOwner,
+            { response ->
+                try {
+                    AuthRepository(sessionRepository).authProvider(response)
+                    view.findNavController().navigate(R.id.homeFragment)
+                    setUpBottomNavigationItem()
+                } catch (t: IllegalStateException) {
+                    showErrorToast(t)
+                }
+            }
+        )
+    }
+
+    private fun <T> showErrorToast(t: T) {
+        when (t) {
+            is AlreadyAuthorizedException -> {
+                Toast.makeText(
+                    context,
+                    resources.getString(R.string.error_already_authorized),
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+            is InvalidDataException -> {
+                Toast.makeText(
+                    context,
+                    resources.getString(R.string.error_invalid_data),
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+            is WrongPasswordException -> {
+                Toast.makeText(
+                    context,
+                    resources.getString(R.string.error_wrong_password),
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+            is UserNotFoundException -> {
+                Toast.makeText(
+                    context,
+                    resources.getString(R.string.error_user_not_found),
+                    Toast.LENGTH_SHORT
+                ).show()
```

Comment:
wdyt about moving part of this logic into ViewModel? e.g. mapping from exception to string res id?
It's better to keep fragments as simple as possible, without non presentation logic